### ISSUE TITLE
the loss became nan after a few train steps

### ISSUE THREAD

**Issue by boya34** [lucidrains/gigagan-pytorch#52]
My dataset is about 1w images of someone wearing different cloths, and the text is the captions of the images.
![image](https://github.com/lucidrains/gigagan-pytorch/assets/60995486/74b09da0-ab2e-4fb3-b5ae-9a386fa6aac6)
![image](https://github.com/lucidrains/gigagan-pytorch/assets/60995486/c584d970-7517-4f56-9d01-a76e2d53d1ab)
but the loss became so strange...
![image](https://github.com/lucidrains/gigagan-pytorch/assets/60995486/7d1b61b3-a7ef-4d81-a5f3-7f0942a72458)
does anyone know how does it happen?

**Comment by boya34** [comment]
this is my train code
`import torch

from gigagan_pytorch import (
    GigaGAN,
    ImageDataset, TextImageDataset
)
from safetensors import safe_open

gan = GigaGAN(
    generator = dict(
        dim_capacity = 8,
        style_network = dict(
            dim = 64,
            depth = 4,
            dim_text_latent = 64
        ),
        image_size = 512,
        dim_max = 512,
        num_skip_layers_excite = 4,
        unconditional = False,
        text_encoder = dict(
            dim = 64,
            depth = 4
        )
    ),
    discriminator = dict(
        dim_capacity = 16,
        dim_max = 512,
        image_size = 512,
        num_skip_layers_excite = 4,
        unconditional = False,
        text_encoder = dict(
            dim = 64,
            depth = 4
        )
    ),
    amp = True,
    save_and_sample_every = 500,
    model_folder = './gigagan-models/***',
    results_folder = './gigagan-results/***',
).cuda()

# dataset

# dataset = ImageDataset(
#     folder = './test_data/',
#     image_size = 256
# )

dataset = TextImageDataset(
    folder = '***',
    image_size = 512
)

dataloader = dataset.get_dataloader(batch_size = 1)

# you must then set the dataloader for the GAN before training

gan.set_dataloader(dataloader)

# training the discriminator and generator alternating
# for 100 steps in this example, batch size 1, gradient accumulated 8 times

gan(
    steps = 10000,
    grad_accum_every = 8
)
gan.save('gigagan-models/***/final.ckpt')
print("model saved!")
# after much training

`

**Comment by unk4now75** [comment]
Did you fix it?
